src/main/java/JsonIO.java

```diff
@@ -30,15 +33,75 @@ void writeToJson(String pathToJson, Object object) {
 
     Snippet loadSnippetFromJson(String pathToJson) {
         try {
-            return g.fromJson(new FileReader(pathToJson), Snippet.class);
+            return gson.fromJson(new FileReader(pathToJson), Snippet.class);
         } catch (FileNotFoundException ex) {
             onException(ex);
             return null;
         }
     }
 
+    List<Snippet> parseSnippetsFromGistsApi(String allSnippetsData) {
+        List<Snippet> parsedSnippets = new ArrayList<Snippet>();
+
+        JsonArray gistArray = gson.fromJson(allSnippetsData, JsonArray.class);
+
+        for (int i = 0; i < gistArray.size(); i++) {
+            String gistUrl = gistArray.get(i).getAsJsonObject().get("id").getAsString();
+            Snippet fetchedSnippet = GistsApi.getInstance().getSpecificSnippet(gistUrl);
+            parsedSnippets.add(fetchedSnippet);
+        }
+
+        return parsedSnippets;
+    }
+
+    Snippet parseSnippetAttributesFromGistsApi(String snippetData) {
+        JsonObject completeGistObject = gson.fromJson(snippetData, JsonObject.class);
+        JsonObject relevantGistData = completeGistObject
+                .get("files").getAsJsonObject() // Gist object contains files
+                .entrySet().iterator().next() // But we only need one file, the first one
+                .getValue().getAsJsonObject(); // This json object contains all the relevant data
+
+        Snippet parsedSnippet = new Snippet(
+                null,
```

Comment:
Perhaps add a small comment on what this is. The rest of the parameters can be explained by their content, but "null" doesn't really say much – and you have to look it up again.
Might be nitpicky of me, but saves time later.